packages/core/interfaces/src/index.ts

```diff
@@ -147,16 +147,20 @@ export interface UseLocale
   LOCALE,
   COUNTRY,
   CURRENCY,
-  AVAILABLE_LOCALES,
-  AVAILABLE_COUNTRIES,
-  AVAILABLE_CURRENCIES,
+  AVAILABLE_LOCALES = LOCALE[],
+  AVAILABLE_COUNTRIES = COUNTRY[],
+  AVAILABLE_CURRENCIES = CURRENCY[],
 > {
-  locale: LOCALE;
-  country: COUNTRY;
-  currency: CURRENCY;
-  availableLocales: AVAILABLE_LOCALES;
-  availableCountries: AVAILABLE_COUNTRIES;
-  availableCurrencies: AVAILABLE_CURRENCIES;
+  locale: ComputedProperty<LOCALE>;
+  country: ComputedProperty<COUNTRY>;
+  currency: ComputedProperty<CURRENCY>;
+  setLocale: (locale: LOCALE) => void;
+  setCountry: (country: COUNTRY) => void;
+  setCurrency: (currency: CURRENCY) => void;
```

Comment:
I'd add promises here to cover cases, where integration may store these pieces of information on API
```suggestion
  setLocale: (locale: LOCALE) => Promise<void>;
  setCountry: (country: COUNTRY) => Promise<void>;
  setCurrency: (currency: CURRENCY) => Promise<void>;
```